contacts.py

```diff
@@ -37,10 +37,13 @@ def dictionary_factory():
         return defaultdict(FSBackedThreeLevelDict.dictionary_factory)
 
     def __init__(self, directory):
+        # { AA: { BB: { CC: AABBCCDEF123: (floatingseconds, serialnumber) } } }
```

Comment:
missing brackets:

 { AA: { BB: { CC: AABBCCDEF123: [(floatingseconds_1, serialnumber_1),  (floatingseconds_2, serialnumber_2), ...]} }

Reply:
Thanks - that's why I did this :-)